runtime/lua/vim/ui.lua

```diff
@@ -44,7 +44,7 @@ function M.select(items, opts, on_choice)
   opts = opts or {}
   local choices = { opts.prompt or 'Select one of:' }
   local format_item = opts.format_item or tostring
-  for i, item in pairs(items) do
+  for i, item in ipairs(items) do
```

Comment:
Just to double check: The function didn't work with non-list tables before?

Reply:
So I had initially added this because LuaLS couldn't infer the type of `i`, but when reading the doc and code I figured out that `ipairs` was better suited here. After changing the type of `items` from `table` to `any[]` then `pairs` doesn't generate a warning anymore, but I still think `ipairs` is intended here?